assignments/lambda-classes.js

```diff
@@ -1 +1,174 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(data) {
+        this.name = data.name;
+        this.age = data.age;
+        this.location = data.location;
+        this.gender = data.gender;
+    }
+
+    speak() {
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}.`)
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructorData) {
+        super(instructorData);
+        this.specialty = instructorData.specialty;
+        this.favLanguage = instructorData.favLanguage;
+        this.catchPhrase = instructorData.catchPhrase;
+    }
+
+    demo(subject) {
+        console.log(`Today we are learning about ${subject}.`);
+    }
+    grade(student, subject) {
+        console.log(`${student.name} receives a perfect score on ${subject}.`)
+    }
+    gradeAssignment(student) {
+        return student.grade = ((Math.floor(Math.random() * (1 - 0 + 1)) + 0) ? student.grade + 5 : student.grade - 5);
+    }
+}
+
+class Student extends Person {
+    constructor(studentData) {
+        super(studentData);
+        this.previousBackground = studentData.previousBackground;
+        this.className = studentData.className;
+        this.favSubjects = studentData.favSubjects;
+        this.grade = studentData.grade;
+    }
+
+    listsSubjects() {
+        this.favSubjects.forEach(function(subject) {
+            console.log(subject);
+        });
+    }
+    PRAssignment(subject) {
+        console.log(`${this.name} has submitted a PR for ${subject}`);
+    }
+    sprintChallenge(subject) {
+        console.log(`${this.name} has begun sprint challenge on ${subject}.`);
+    }
+    graduate() {
+        console.log(((this.grade >= 70)? `${this.name} has graduated!!!` : `${this.name} must have more assignments graded to graduate. Grade: ${this.grade}`));
```

Comment:
If it helps, you can format ternary operators in different ways. For example.
*this.grade >= 70 
     ? code here
      : code here ;*
or
*this.grade >= 70 ? (
    code here
) : (
    code here
);*